awx/main/scheduler/task_manager.py

```diff
@@ -451,6 +461,8 @@ def process_pending_tasks(self, pending_tasks):
             if self.is_job_blocked(task):
                 logger.debug("{} is blocked from running".format(task.log_format))
                 continue
+            if self.start_task_limit <= 0:
+                continue
```

Comment:
wouldn't it be better to put this before `self.is_job_blocked`?

Or even, if you hit the condition, `break` instead of `continue`

Reply:
yes you're right, can `break` and be before `is_job_blocked`